COCOAPODS.md

```diff
@@ -1,150 +1,343 @@
-of an Xcode project. After that the framework built from the Kotlin/Native module can be used from
-this Xcode project. If necessary, this framework is automatically rebuilt during Xcode build process.
+2. Re-import the project.
 
-## Workflow
+You can find a sample project [here](https://github.com/zoldater/KotlinWithLocalObjCPods). 
+ 
+## Use a Kotlin Gradle project as a CocoaPods dependency
 
-To import a Kotlin/Native module in an existing Xcode project:
+You can use a Kotlin/Native Gradle project as a CocoaPods dependency (Kotlin Pod). Such a representation provides the
+ following advantages:
 
-0. Make sure that you have CocoaPods [installed](https://guides.cocoapods.org/using/getting-started.html#installation).
-We recommend using CocoaPods 1.6.1 or later.
+ * Such a dependency can be included in a `Podfile` of an Xcode project and automatically built (and rebuilt)
+ along with this project. It simplifies importing to Xcode by removing a need to write corresponding Gradle tasks and
+  Xcode build steps manually.
+ * When building from Xcode, you can use CocoaPods libraries without writing
+ `.def` files manually and setting `cinterop` tool parameters. In this case, all required parameters can be
+ obtained from the Xcode project configured by CocoaPods.
+ 
+You can add dependencies between:
+* [A Kotlin Pod and an Xcode project with one target](#add-a-dependency-between-a-kotlin-pod-and-xcode-project-with-one-target)
+* [A Kotlin Pod and an Xcode project with several targets](#add-a-dependency-between-a-kotlin-pod-with-an-xcode-project-with-several-targets)
+* [Several Kotlin Pods and an Xcode project](#add-dependencies-between-multiple-kotlin-pods-and-an-xcode-project)
 
-1. Configure a Gradle project: apply the `org.jetbrains.kotlin.native.cocoapods` plugin, add
-and configure the targets, and specify the required podspec fields.
+> To correctly import the dependencies into the Kotlin/Native module, the
+`Podfile` must contain either [`use_modular_headers!`](https://guides.cocoapods.org/syntax/podfile.html#use_modular_headers_bang)
+or [`use_frameworks!`](https://guides.cocoapods.org/syntax/podfile.html#use_frameworks_bang)
+directive.
+{:.note}
 
-2. Run the `podspec` task. The podspec file described above will be generated.
+### Add a dependency between a Kotlin Pod and Xcode project with one target
 
-    > In order to avoid compatibility issues during an Xcode build, the plugin requires using
-    a [Gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html). 
-    To generate the wrapper automatically during execution of the `podspec` task,
-    run it with the parameter `-Pkotlin.native.cocoapods.generate.wrapper=true`.
+1. Create an Xcode project with `Podfile` if you haven’t done this yet.
+2. Add the path to your `Podfile` with `podfile = projectDir.resolve("..")` to `build.gradle.kts` (`build.gradle`) 
```

Comment:
(minor) It's better to replace `podfile = projectDir.resolve("..")` with `podfile = project.file(..)`. The second construction is more common in Gradle.

Reply:
@zoldater Should we update sample projects and docs?

Reply:
I prefer this syntax because it shows more implicitly that entered path  is related to `projectDir`. As these 2 constructions are semantically equivalent, we can replace it.

Reply:
Updated the syntax - replaced `podfile = projectDir.resolve("..")` with `podfile = project.file(..)`

Reply:
@zoldater will update sample projects.

Reply:
updated links to sample projects.